indexed_gzip/zran.h

```diff
@@ -304,4 +304,74 @@ int64_t zran_read(
   uint64_t       len    /* Number of bytes to read   */
 );
 
+/* Return codes for zran_export_index. */
+enum {
+    ZRAN_EXPORT_OK          =  0,
+    ZRAN_EXPORT_WRITE_ERROR = -1
+};
+
+/*
+ * Export current index data to given file. This exported file later can be
+ * used to rebuild index without needing to going through the file again.
+ *
+ * See zran_import_index for importing.
+ *
+ * Returns:
+ *   - ZRAN_EXPORT_OK for success.
+ *
+ *   - ZRAN_EXPORT_WRITE_ERROR to indicate an error from writing to underlying
+ *     file.
+ */
+int zran_export_index(
+  zran_index_t  *index; /* The index                  */
+  FILE          *fd;    /* Open handle to export file */
+);
+
+/* Return codes for zran_import_index. */
+enum {
+    ZRAN_IMPORT_OK           =  0,
+    ZRAN_IMPORT_FAIL         = -1,
+    ZRAN_IMPORT_EOF          = -2,
+    ZRAN_IMPORT_READ_ERROR   = -3,
+    ZRAN_IMPORT_OVERFLOW     = -4,
+    ZRAN_IMPORT_INCONSISTENT = -5,
+    ZRAN_IMPORT_MEMORY_ERROR = -6
+};
+
+/*
+ * Import current index from the given file. Existing index will be deleted.
```

Comment:
The implementation currently does not explicitly support importing into an uninitialised `zran_index_t` struct, e.g.:

```c
zran_index_t index;
zran_import_index(&index);
```

Could you make a note in the docs that the index must have been initialised via `zran_init`?

Reply:
Sure. I will also add that `spacing` and `window_size` will be overwritten even if they are initialized, while `readbuf_size` will be kept.